docs/source/securing-calico.rst

```diff
@@ -78,9 +78,99 @@ traffic that should be allowed.
 Securing etcd
 -------------
 
+Limiting network access to etcd
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
 Calico uses etcd to store and forward the configuration of the network from
 plugin to the Felix agent.  By default, etcd is writable by anyone with
 access to its REST interface.  We plan to use the RBAC feature of an upcoming
 etcd release to improve this dramatically.  However, until that work is done,
 we recommend blocking access to etcd from all but the IP range(s) used by the
 compute nodes and plugin.
+
+.. _usingtlswithetcd:
+
+Using TLS to encrypt and authenticate communication with etcd
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Calico supports etcd's TLS-based security model, which supports the encryption
+(and authentication) of traffic between Calico components and the etcd cluster.
+
+.. warning:: Calico's TLS support uses the Python urllib3 library.
+             Unfortunately, we've seen issues with TLS in some of the common
+             versions of urllib3 including the version that ships with
+             Ubuntu 14.04 (1.7.1) and versions 1.11-1.12.  Version 1.13
+             appears to work well.
+
+To enable TLS support:
+
+* Follow the instructions in the `etcd security guide`_ to create a certificate
+  authority and enable TLS in etcd.  We recommend enabling both client and
+  peer authentication.  This will enable security between Calico and etcd as
+  well as between different nodes in the etcd cluster.
+
+  .. note:: etcd proxies communicate with the cluster as peers so they need to
+            have peer certificates.
+
+* Issue a private key and client certificate for each Calico component.  In
+  OpenStack, you'll need one certificate for each control node and one for
+  each compute node.
+
+* If using OpenStack, configure each Neutron control node to use TLS:
+
+  * Put the PEM-encoded private key and certificates in a secure place that is
+    only accessible by the user that neutron will run as.
+
+    .. note:: each control node should have its own private key and
+              certificate.  The certificate encodes the IP address or
+              hostname of the owner.
+
+  * Modify ``/etc/neutron/plugins/`` to include a ``[calico]`` section that
```

Comment:
Did you mean to say ml2_conf.ini here?  Or are you anticipating that we won't be ML2 any more after the floating IP change?